ansible/roles/test/files/ptftests/vxlan-decap.py

```diff
@@ -0,0 +1,293 @@
+
+            test['vlan_ip_prefix'] = '.'.join(gw.split('.')[0:3])+".%d"
+
+            self.tests.append(test)
+
+        self.dut_mac = graph['dut_mac']
+
+        ip = None
+        for data in graph['minigraph_lo_interfaces']:
+            if data['prefixlen'] == 32:
+                ip = data['addr']
+                break
+        else:
+            raise Exception("ipv4 lo interface not found")
+
+        self.loopback_ip = ip
+
+        self.ptf_mac_addrs = self.readMacs()
+
+        self.generate_ArpResponderConfig()
+
+        self.cmd(["supervisorctl", "start", "arp_responder"])
+
+        self.dataplane.flush()
+
+        return
+
+    def tearDown(self):
+        self.cmd(["supervisorctl", "stop", "arp_responder"])
+
+        return
+
+    def runTest(self):
+        print
+        for test in self.tests:
+            print test['name']
+            res_v = self.Vxlan(test)
+            print "  Vxlan  = ", res_v
+            res_f = self.FromPC(test)
+            print "  FromPC = ", res_f
+            res_t = self.ToPC(test)
+            print "  ToPC   = ", res_t
+            print
+            if self.vxlan_enabled:
+                self.assertTrue(res_v, "VxlanTest failed")
+            else:
+                self.assertFalse(res_v, "VxlanTest must be disabled")
+            self.assertTrue(res_f, "FromPC test failed")
+            self.assertTrue(res_t, "ToPC test failed")
+
+    def Vxlan(self, test):
+        rv = True
+        for n in self.net_ports:
+            for a in test['acc_ports']:
+                res = self.checkVxlan(a, n, test)
+                rv = rv and res
+
+        return rv
+
+    def FromPC(self, test):
+        rv = True
+        for n in self.net_ports:
+            for a in test['acc_ports']:
+                res = self.checkRegularFromPC(a, n, test)
+                rv = rv and res
+        return rv
+
+    def ToPC(self, test):
+        rv = True
+        for dst, ports in self.pc_info:
+            for a in test['acc_ports']:
+                res = self.checkRegularToPC(a, ports, dst, test)
+                rv = rv and res
+        return rv
+
+    def checkRegularToPC(self, acc_port, pc_ports, dst_ip, test):
+        rv = True
+        src_mac = self.ptf_mac_addrs['eth%d' % acc_port]
+        dst_mac = self.dut_mac
+        src_ip = test['vlan_ip_prefix'] % acc_port
+
+        packet = simple_tcp_packet(
+                         eth_dst=dst_mac,
+                         eth_src=src_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                       )
+        exp_packet = simple_tcp_packet(
+                         eth_dst=self.random_mac,
+                         eth_src=dst_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                         ip_ttl = 63,
+                       )
+
+        exp_packet = Mask(exp_packet)
+        exp_packet.set_do_not_care(0, 6*8)
```

Comment:
Can you please use `exp_packet.set_do_not_care_scapy(scapy.Ether, "dst")` instead, if that is what is intended?